config/jupyterhub_config.py

```diff
@@ -4,24 +4,40 @@
 https://jupyterhub.readthedocs.io/en/latest/tutorial/getting-started/config-basics.html
 https://jupyterhub.readthedocs.io/en/stable/reference/config-reference.html
 """
+
 import os
 
+import nativeauthenticator
+
 from jupyterhub_config.custom_spawner import VirtualEnvSpawner
 
 c = get_config()
 
-# Set the authenticator class
-# TODO: Change the authenticator class to a secure one (e.g. GitHubOAuthenticator)
-c.JupyterHub.authenticator_class = 'jupyterhub.auth.DummyAuthenticator'
-c.Authenticator.allowed_users = {'spark_user', 'test_user1', 'test_user2'}
-c.DummyAuthenticator.password = os.environ['JUPYTERHUB_ADMIN_PASSWORD']
+# Set the authenticator class to nativeauthenticator
+# ref: https://native-authenticator.readthedocs.io/en/latest/quickstart.html
+c.JupyterHub.authenticator_class = 'native'
+c.JupyterHub.template_paths = [f"{os.path.dirname(nativeauthenticator.__file__)}/templates/"]
+c.NativeAuthenticator.open_signup = True
+c.NativeAuthenticator.check_common_password = True
+c.NativeAuthenticator.minimum_password_length = 6
 
+# Set up the admin user
 c.Authenticator.admin_users = {'spark_user'}
+# TODO set admin user password to os.environ['JUPYTERHUB_ADMIN_PASSWORD'] automatically - currently spark_user is created manually with the signup page
+c.Authenticator.allow_all = True
+
+c.JupyterHub.cookie_secret_file = f"{os.environ['JUPYTERHUB_USER_HOME']}/jupyterhub_cookie_secret"
+c.JupyterHub.db_url = f"sqlite:///{os.environ['JUPYTERHUB_USER_HOME']}/jupyterhub.sqlite"
```

Comment:
If this is the user database it probably shouldn't be stored in the user home dir

Reply:
where would we store it?

Reply:
I think that would be the same, volume is mounted, whether it's `/jupyterhubsecrets` or `/jupyterhub/users_home`. Users know where to find it, users can see it. But no r/w access.

Reply:
wait users don't have r/w to JUPYTERHUB_USER_HOME ?

Reply:
no. they have w/r to their own dir. But not `JUPYTERHUB_USER_HOME`

Reply:
Isn't that the root of their home dirs?

Reply:
It is. But that dir belongs to `spark_user`

Reply:
this will also be in `jupyterhub_secrets` dir

Reply:
:+1: